apache-http-client/src/main/java/com/joyent/http/signature/apache/httpclient/HttpSignatureAuthScheme.java

```diff
@@ -54,6 +68,13 @@
     private final ThreadLocalSigner signer;
 
     /**
+     * Map of credentials to cache object used for looking up cached signatures.
+     */
+    private ConcurrentMap<Credentials, HttpSignatureCache> signatureCacheMap =
```

Comment:
`HttpAuthScheme` can be using in two ways. It can be used as via an interceptor or it can be using via the authorization API as provided with the Apache HTTP Client. It is possible in the latter case, to have multiple credentials being used against a single `HttpAuthScheme`.

I'm not too worried about the cache being unbounded. I think that once you get to memory sizes that are problematic, you are more than likely going to run into problems with Triton handling too many users. Credentials is a small object and a ConcurrentHashMap should be able to handle 500k entries with ease.

The only use case I can imagine where it may be problematic is if an implementor is passing on user-supplied credentials. However, The lifecycle of the Map is tied to the lifecycle of a `MantaClient` in that `MantaClient` → `MantaConnectionFactory` → `MantaConnectionFactory.createConnection()` → `MantaConnectionFactory.createBuilder` → `builder.addInterceptorLast(new HttpSignatureRequestInterceptor(...))` → `HttpSignatureRequestInterceptor.authScheme` → `HttpSignatureAuthScheme.signatureCacheMap`

Essentially, one `MantaClient` will yield one signature cache map with their lifecycles tied. I would expect that applications would want to eventually close their `MantaClient` instances and there by free up the Map.

We could use any number of bounded data structures to mitigate this problem. However, I don't want to bring in extra external libraries unless I have to for this library. My preference would be to wait to see if this is actually a problem for anyone.